### PR TITLE
Set GOMAXPROCS for etcd

### LINKED ISSUE #7767: Slow etcd
After recent performance fixes we are now not bound by CPU or disk IOPS. Apiserver during density test uses around 80% of CPU. Still we observe high request latency for pods, events, endpoints and sometimes other resources. I was able to narrow this down to slow etcd.

Etcd uses up to 80% of CPU. For some reason we see high latency of requests to etcd. Increasing disk (to increase IOPS) or number of cores doesn't help. In order to meet criteria defined in 1.0 roadmap (#4521) we have to make it much faster.

@xiang90 Do you have some suggestions how to debug this? Any ideas what could be the reason?

@wojtek-t @davidopp @xiang90

### PR DESCRIPTION
**Pull request by fgrzadkowski**
This fixes #7767

@timothysc @wojtek-t

### CODE REVIEW

File: cluster/saltbase/salt/etcd/etcd.manifest [hunk 1]
```diff
@@ -1,3 +1,5 @@
+{% set num_nodes = salt['grains.get']('num_cpus', '') %}
```
Comment (lavalamp) [inline cluster/saltbase/salt/etcd/etcd.manifest]: num_cpus as a name maybe?

**Comment by wojtek-t** [comment]
LGTM

File: cluster/saltbase/salt/etcd/etcd.manifest (hunk 1 above)
Comment (fgrzadkowski) [inline cluster/saltbase/salt/etcd/etcd.manifest]: Oops :) I copy&pasted this line from an example and forgot to change the variable name. Thanks!

**Comment by timothysc** [comment]
kind of sledgehammering it, ideally it would be good to know optimal values.

**Comment by lavalamp** [comment]
@timothysc optimal value probably depends on the number of cores & workload; this seems like a good guess. When I've tried optimizing GOMAXPROCS for other go programs, number of cpus+1 seemed the best. But etcd could be very different.

**Pull request merged**